### PR TITLE
Sprite offset and texture mapping

### PR DESCRIPTION
**Pull request by GDQR**
Allows you to create an offset in the sprite and show only a part of the original image, useful for sprite sheets.
Allows you to select the texture mapping (Nearest or Linear) for renderer 2D.
Includes a tutorial.

### CODE REVIEW

File: tutorials/10-sprite-sheet/inc/fontSprite.hpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/*
+# _____        ____   ___
+#   |     \/   ____| |___|
+#   |     |   |   \  |   |
+#-----------------------------------------------------------------------
+# Copyright 2022, tyra - https://github.com/h4570/tyra
+# Licensed under Apache License 2.0
+# Sandro Sobczyński <sandro.sobczynski@gmail.com>
```
Comment (Wolf3s) [inline tutorials/10-sprite-sheet/inc/fontSprite.hpp]: Add your name but keep sandro and replace copyright data to 2022-2023 on every file tutorial header and source file.

**Review by Wolf3s — CHANGES_REQUESTED** [review]

File: tutorials/10-sprite-sheet/inc/fontSprite.hpp (hunk 1 above)
Comment (Wolf3s) [reply tutorials/10-sprite-sheet/inc/fontSprite.hpp]: Also can you do clang format?

Comment (GDQR) [reply tutorials/10-sprite-sheet/inc/fontSprite.hpp]: Well I'll change that, but what is that clang format?

Comment (GDQR) [reply tutorials/10-sprite-sheet/inc/fontSprite.hpp]: for the tutorial you need this images 
atlas
earthbound-Font

**Comment by h4570** [comment]
Hello!

Thank you so much for contributing.
So.. from now, we have font support in Tyra 😄.
I added some fixes to your PR, **please check if I didn't broke anything** and respond to this comment.
After your confirmation, I will merge this PR and add new assets to release zip file

File: tutorials/10-sprite-sheet/src/font_sprite.cpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+# Copyright 2022-2023, tyra - https://github.com/h4570/tyra
+# Licensed under Apache License 2.0
+# Guido Diego Quispe Robles
+*/
+
+#include "font_sprite.hpp"
+
+namespace Tyra {
+
+const int Font::chars[FONT_CHAR_SIZE]{
```
Comment (h4570) [inline tutorials/10-sprite-sheet/src/font_sprite.cpp:15]: I changed some letters, because of:

image

**Comment by GDQR** [comment]
XD I was just trying to make a commit, I just found out what the clang format is, I had some problems but don't worry I already fixed it.
By the way, I made the font simple and so that it would only be used for this tutorial and give others an idea to create their own font, that's why I put it in a class, otherwise I would have made it global, it doesn't bother me anyway.

**Comment by GDQR** [comment]
> Hello!
> 
> Thank you so much for contributing. So.. from now, we have font support in Tyra 😄. I added some fixes to your PR, **please check if I didn't broke anything** and respond to this comment. After your confirmation, I will merge this PR and add new assets to release zip file

I'm quoting this right now XD, sorry, I'm new to contributing projects.

**Comment by Wolf3s** [comment]
> > Hello!
> > Thank you so much for contributing. So.. from now, we have font support in Tyra 😄. I added some fixes to your PR, **please check if I didn't broke anything** and respond to this comment. After your confirmation, I will merge this PR and add new assets to release zip file
> 
> I'm quoting this right now XD, sorry, I'm new to contributing projects.

Ok, at least you can learn here.

File: tutorials/10-sprite-sheet/src/font_sprite.cpp (hunk 2 above)
Comment (GDQR) [reply tutorials/10-sprite-sheet/src/font_sprite.cpp:15]: It's fine, no problem

**Comment by GDQR** [comment]
> > > Hello!
> > > Thank you so much for contributing. So.. from now, we have font support in Tyra smile. I added some fixes to your PR, **please check if I didn't broke anything** and respond to this comment. After your confirmation, I will merge this PR and add new assets to release zip file
> > 
> > 
> > I'm quoting this right now XD, sorry, I'm new to contributing projects.
> 
> Ok, at least you can learn here.

yeah, i hope i can contribute more to tyra.

**Comment by wolfysdl** [comment]
> 
lgtm

**Comment by Wolf3s** [comment]
> Hello!
> 
> Thank you so much for contributing. So.. from now, we have font support in Tyra 😄. I added some fixes to your PR, **please check if I didn't broke anything** and respond to this comment. After your confirmation, I will merge this PR and add new assets to release zip file

Lgtm commit.

**Comment by Wolf3s** [comment]
> > > > Hello!
> > > > Thank you so much for contributing. So.. from now, we have font support in Tyra smile. I added some fixes to your PR, **please check if I didn't broke anything** and respond to this comment. After your confirmation, I will merge this PR and add new assets to release zip file
> > > 
> > > 
> > > I'm quoting this right now XD, sorry, I'm new to contributing projects.
> > 
> > 
> > Ok, at least you can learn here.
> 
> yeah, i hope i can contribute more to tyra.

Be welcome.

**Review by h4570 — APPROVED** [review]

**Pull request merged**